Don't replace CMake's find_package

- [x] Checked for duplicates

### Describe the bug

ROOT overrides CMake's `find_package` in `cmake/modules/SearchInstalledSoftware.cmake`, which is not uncommon, but relies on undocumented behavior of CMake. See e.g.: https://crascit.com/2018/09/14/do-not-redefine-cmake-commands/

This however breaks other tools, that also rely on this behavior, resulting in an infinite recursion. My concrete example is vcpkg, a popular package manager (not only for Windows), that pulls itself into a CMake project via a toolchain file to inject dependencies. Unfortunately, vcpkg needs to hook `find_package` itself to point it to the installed dependencies.

### Expected behavior

ROOT can be configured with vcpkg and other tools sensitive to overriden `find_package`. This requires ROOT to not override `find_package`.

### To Reproduce

Clone ROOT, create `build_win` folder, run:
```
PS C:\dev\root\build_win> cmake -DCMAKE_TOOLCHAIN_FILE=D:\dev\vcpkg\scripts\buildsystems\vcpkg.cmake ..
-- Building for: Visual Studio 16 2019
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19042.
-- The C compiler identification is MSVC 19.29.30037.0
-- The CXX compiler identification is MSVC 19.29.30037.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30037/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30037/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: C:/Program Files/Git/cmd/git.exe (found version "2.30.0.windows.2")
-- Detected ROOT_VERSION 6.25.01
-- Looking for Python
-- Could NOT find Python3 (missing: Python3_NumPy_INCLUDE_DIRS NumPy) (found version "3.9.2")
-- Could NOT find Python2 (missing: Python2_EXECUTABLE Python2_LIBRARIES Python2_INCLUDE_DIRS Python2_NumPy_INCLUDE_DIRS Interpreter Development NumPy Development.Module Development.Embed)
    Reason given by package:
        Interpreter: Wrong major version for the interpreter "C:/Program Files/Python39/python.exe"

-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- Performing Test GLIBCXX_USE_CXX11_ABI
-- Performing Test GLIBCXX_USE_CXX11_ABI - Failed
-- ROOT Platform: win32
-- ROOT Compiler: MSVC 19.29.30037.0
-- ROOT Processor: AMD64
-- ROOT Architecture: win32
-- Build Type: '' (flags = '')
-- Compiler Flags: -nologo -IC:/dev/root/build/win -FIw32pragma.h -FIsehmap.h -Zc:__cplusplus -MD -GR -EHsc- -W3 -wd4141 -wd4291 -wd4244 -wd4049 -D_WIN32  -D_XKEYCHECK_H -DNOMINMAX -D_CRT_SECURE_NO_WARNINGS
-- ROOT default compression algorithm: zlib
-- PyROOT will be built for version 3.9.2
-- Looking for ZLib
CMake Error at D:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake:693 (set):
  Maximum recursion depth of 1000 exceeded
Call Stack (most recent call first):
  D:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake:771 (_find_package)
... // many stack frames
  D:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake:771 (_find_package)
  cmake/modules/SearchInstalledSoftware.cmake:16 (_find_package)
  cmake/modules/SearchInstalledSoftware.cmake:55 (find_package)
  CMakeLists.txt:220 (include)
```

### Setup

ROOT master cloned today on Windows 10.

--- a/cmake/modules/SearchInstalledSoftware.cmake
+++ b/cmake/modules/SearchInstalledSoftware.cmake
@@ -13,12 +13,6 @@ include(FindPackageHandleStandardArgs)
 set(lcgpackages http://lcgpackages.web.cern.ch/lcgpackages/tarFiles/sources)
 string(REPLACE "-Werror " "" ROOT_EXTERNAL_CXX_FLAGS "${CMAKE_CXX_FLAGS} ")
 
-macro(find_package)
-  if(NOT "${ARGV0}" IN_LIST ROOT_BUILTINS)
-    _find_package(${ARGV})
-  endif()
-endmacro()
-
 #---On MacOSX, try to find frameworks after standard libraries or headers------------
 set(CMAKE_FIND_FRAMEWORK LAST)
 
@@ -1985,4 +1979,18 @@ endif()
 # Needed to run tests of the distributed RDataFrame module that use dask.
 if(test_distrdf_dask)
   find_package(Dask 2022.08.1 REQUIRED)
+endif()
+
+#------------------------------------------------------------------------------------
+# Modify find_package to ignore any ROOT builtins, such that the builtin LLVM
+# doesn't find and use system versions of the builtin dependencies. This is a
+# bit hacky, but fortunately this needs to be done only when builtins are used.
+#
+# See also discussion on GitHub: https://github.com/root-project/root/issues/8633
+if(ROOT_BUILTINS)
+  macro(find_package)
+    if(NOT "${ARGV0}" IN_LIST ROOT_BUILTINS)
+      _find_package(${ARGV})
+    endif()
+  endmacro()
 endif()

